src/features/user/UserProfile/components/RedeemModal.tsx

```diff
@@ -27,15 +143,121 @@ export default function RedeemModal({
-        </div>
+
+        {codeRedeemed && validCodeData ? (
+          <>
+            <div className={styles.modalFinal}>
+              <div className={styles.header}>
+                <div onClick={() => closeRedeem()} className={styles.headerCloseIcon}>
+                  <Close />
+                </div>
+                <div className={styles.headerTitle}>
+                  {t('redeem:congratulations')}
+                </div>
+              </div>
+
+              <div className={styles.thankyouImageContainer}>
+                <div className={styles.thankyouImage}>
+                  <div className={styles.thankyouImageHeader}>
+                    <p dangerouslySetInnerHTML={{ __html: t('donate:thankyouHeaderText') }} />
+                  </div>
+                  <div className={styles.donationCount}>
+                    {t('redeem:myPlantedTreesByOrg', { count: validCodeData.treeCount,tpoName:validCodeData.tpos[0].tpoName  })}
+                    <p className={styles.donationTenant}>
+                      {t('donate:plantTreesAtURL', { url: config.tenantURL })}
+                    </p>
+                  </div>
+                </div>
+              </div>
+
+              {/* hidden div for image download */}
+              <div style={{ width: '0px', height: '0px', overflow: 'hidden' }}>
+                <div className={styles.tempThankYouImage} ref={imageRef}>
+                  <div className={styles.tempthankyouImageHeader}>
+                    <p dangerouslySetInnerHTML={{ __html: t('donate:thankyouHeaderText') }} />
+                  </div>
+                  <p className={styles.tempDonationCount}>
+                    {t('redeem:myPlantedTreesByOrg', { count: validCodeData.treeCount,tpoName:validCodeData.tpos[0].tpoName  })}
+                  </p>
+                  <p className={styles.tempDonationTenant}>
+                    {t('donate:plantTreesAtURL', { url: config.tenantURL })}
+                  </p>
+                </div>
+              </div>
+
+              <div className={styles.shareOptions}>
+                <ShareOptions
+                  treeCount={validCodeData.treeCount}
+                  sendRef={sendRef}
+                  handleTextCopiedSnackbarOpen={handleTextCopiedSnackbarOpen}
+                />
+              </div>
+
+              <Snackbar
+                open={textCopiedsnackbarOpen}
+                autoHideDuration={4000}
+                onClose={handleTextCopiedSnackbarClose}
+              >
+                <Alert onClose={handleTextCopiedSnackbarClose} severity="success">
+                  {t('donate:copiedToClipboard')}
+                </Alert>
+              </Snackbar>
+
+            </div>
+          </>
+        ) : (
+            <div className={styles.modal}>
+              {codeValidated && validCodeData ? (
+                <>
+                  <div className={styles.codeTreeCount}>
+                    {validCodeData.treeCount}
```

Comment:
Format this number once this branch and https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/526 is merged!

Reply:
This is just a reminder for myself at https://github.com/Plant-for-the-Planet-org/planet-webapp/issues/549